File: internal/content/testdata/bench/javascript/sequence-double-inverse.js
function sequence_equation(p) {
    let n = p.length;
    let inv = new Array(n + 1).fill(0);
    for (let i = 0; i < n; i++) {
        inv[p[i]] = i + 1;
    }
    let result = [];
    for (let x = 1; x <= n; x++) {
        result.push(inv[inv[x]]);
    }
    return result;
}

File: internal/content/testdata/bench/rust/sequence-double-inverse.rs
fn sequence_equation(p: Vec<i64>) -> Vec<i64> {
    let n = p.len();
    let mut inv = vec![0i64; n + 1];
    for i in 0..n {
        inv[p[i] as usize] = i as i64 + 1;
    }
    (1..=n).map(|x| inv[inv[x] as usize]).collect()
}
